### PR TITLE
feat: add user creation endpoint

### PR DESCRIPTION
**Pull request by HijoDeHonor**
Implemented POST endpoint '/users' for user creation. Added corresponding Express routes and unit tests to verify functionality.

### CODE REVIEW

File: server-with-mock.js
Comment (PiewDev) [inline server-with-mock.js:1]: Is this duplicate file? is equal to that of sql

File: test/user/userControler.test.js
Comment (PiewDev) [inline test/user/userControler.test.js:1]: don't use Magic Strings, uses constants

Comment (PiewDev) [inline test/user/userControler.test.js:1]: Separate the test with comments:

Arrange
Act
Assert

File: test/user/validationUser.test.js
Comment (PiewDev) [inline test/user/validationUser.test.js:1]: don't use Magic Strings, uses constants.

Separate the test with comments:

Arrange
Act
Assert

File: users/mockModel.js
Comment (PiewDev) [inline users/mockModel.js:1]: Is this file necesary?

File: users/mySqlModel.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+}
+const connectionString = DEFAULT_CONFIG
+
+const connection = await mysql.createConnection(connectionString)
+
+export class UserModel {
+  static async create ({ input }) {
+    const { userName, password } = input
+    try {
+      await connection.query(
```
Comment (PiewDev) [inline users/mySqlModel.js:19]: Needs an extra check on whether the user exists

File: users/mySqlModel.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import mysql from 'mysql2/promise'
+import { ERROR_CREATING_USER } from '../utils/textConstants'
+
+const DEFAULT_CONFIG = {
```
Comment (PiewDev) [inline users/mySqlModel.js:4]: Remove these configurations to a separate file

File: server-with-mock.js
Comment (HijoDeHonor) [inline server-with-mock.js:1]: no, this import the UserModel from another file to change if uses a MySQL o local files

File: users/registerUser/userControler.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  constructor ({ userModel }) {
+    this.userModel = userModel;
+  }
+
+  create = async (req, res) => {
+    const { userName, password } = req.body;
+    const result = validateUser(userName, password);
+    if (!result.success) {
+      throw new InvalidDataError(INVALID_DATA, USERS);
+    }
```
Comment (PiewDev) [inline users/registerUser/userControler.js:15]: remove this try-catch should be handled by ErrorHandler

File: users/registerUser/userControler.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    } catch (error) {
+      if (error.message === `${ERROR_CREATING_USER}${USER_ALREADY_EXIST}`) {
+        return res.status(400).json({ error: USER_ALREADY_EXIST });
+      } else {
+        return res.status(500).json({ error: FALIED_CREATE_USER });
+      }
+    }
+  }
+}
+
```
Comment (PiewDev) [inline users/registerUser/userControler.js:30]: Remove these class to a separate file

File: users/userSchema.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+export const validateUser = (userName, password) => {
+  const input = { userName, password }
+
+  return userSchema.safeParse(input);
+}
+
+
+
+
```
Comment (PiewDev) [inline users/userSchema.js]: Remove this comments

File: middlewares/errorHandler/errorhandler.js [hunk 6]
```diff
@@ -1,5 +1,12 @@
+
+
 export const errorHandler = (error, req, res, next) => {
-  const statusCode = error.statusCode || 500
+  const statusCode = Errors[error.name] || 500
   const message = statusCode >= 500 ? 'Internal Server Error' : error.message
   return res.status(statusCode).send(message)
 }
+
```
Comment (PiewDev) [inline middlewares/errorHandler/errorhandler.js]: Are you importing this Constants?

File: server-with-mock.js
Comment (PiewDev) [inline server-with-mock.js:1]: Is this duplicate file? is equal to that of sql

File: test/user/userControler.test.js
Comment (PiewDev) [inline test/user/userControler.test.js:1]: don't use Magic Strings, uses constants

Separate the test with comments:

Arrange
Act
Assert

File: test/user/validationUser.test.js
Comment (PiewDev) [inline test/user/validationUser.test.js:1]: Separate the test with comments:

Arrange
Act
Assert

File: users/mockModel.js
Comment (PiewDev) [inline users/mockModel.js:1]: Is this file necesary?

File: users/mySqlModel.js (hunk 2 above)
Comment (PiewDev) [inline users/mySqlModel.js]: Remove these configurations to a separate file

File: users/mySqlModel.js (hunk 1 above)
Comment (PiewDev) [inline users/mySqlModel.js]: Needs an extra check on whether the user exists

File: users/registerUser/userControler.js (hunk 3 above)
Comment (PiewDev) [inline users/registerUser/userControler.js:15]: remove this try-catch should be handled by ErrorHandler

File: users/registerUser/userControler.js (hunk 4 above)
Comment (PiewDev) [inline users/registerUser/userControler.js]: Remove these class to a separate file

**Review by PiewDev — CHANGES_REQUESTED** [review]
It's pretty good, there are only small things to change.

File: package-lock.json
Comment (PiewDev) [inline package-lock.json:1]: package.lock.json should not be saved to the repository

File: package.json [hunk 7]
```diff
@@ -16,11 +18,16 @@
...
-    "extends": "standard"
+    "extends": "./node_modules/standard/eslintrc.json"
   },
   "dependencies": {
     "dotenv": "16.4.5",
     "express": "4.19.2",
-    "uuid": "10.0.0"
+    "fs": "^0.0.1-security",
+    "mysql2": "^3.10.1",
+    "sinon": "^18.0.0",
```
Comment (PiewDev) [inline package.json]: In the end you didn't use Sinon

File: test/user/userControler.test.js [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+      .mockImplementationOnce(() => Promise.resolve({ userName: uniqueUserName }));
+
+    // Act
+    const res = await request(app)
+      .post('/users')
+      .send({ userName: uniqueUserName, password: TEST_PASSWORD });
+
+    // Assert
+    expect(res.status).toBe(201);
+    expect(res.body).toEqual(uniqueUserName);
```
Comment (PiewDev) [inline test/user/userControler.test.js]: The response from the endpoint should be body.userName. make the change in the controller

File: .vscode/settings.json
Comment (PiewDev) [inline .vscode/settings.json:1]: settings.json should not be saved to the repository

**Review by PiewDev — CHANGES_REQUESTED** [review]
Very good, we need to make some small changes related to the health of the repository.

File: .vscode/settings.json
Comment (PiewDev) [inline .vscode/settings.json:1]: Add to gitignore

File: users/registerUser/userControler.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+  }
+  create = tryCatch(async (req, res) => {
+    const { userName, password } = req.body
+    const result = validateUser(userName, password);
+    if (!result.success) {
+      throw new InvalidDataError(INVALID_DATA, USERS);
+    }
+    const newUser = await this.userModel.create({ input: result.data });
+    if (newUser) {
+      res.status(201).json({ userName: newUser });
```
Comment (PiewDev) [inline users/registerUser/userControler.js]: My bad, the one that should return this format is the model.

**Review by PiewDev — CHANGES_REQUESTED** [review]
A change that you missed from the previous one and a mistake of mine, we are even.

File: users/registerUser/userControler.js (hunk 9 above)
Comment (PiewDev) [reply users/registerUser/userControler.js]: newUser should be an object with the property userName

**Review by PiewDev — APPROVED** [review]
Good job! All the changes look good

**Pull request merged**